liquidctl/driver/aquacomputer.py

```diff
@@ -0,0 +1,258 @@
+                "V",
+            )
+            sensor_readings.append(fan_voltage)
+
+            fan_current = (
+                self._device_info["fan_current_label"][idx],
+                get_unaligned_be16(msg, fan_sensor_offset + _AQC_FAN_CURRENT_OFFSET) * 1e-3,
+                "A",
+            )
+            sensor_readings.append(fan_current)
+
+        # Special-case sensor readings
+        if self._device_info["type"] == self.DEVICE_D5NEXT:
+            # Read +5V voltage rail value
+            plus_5v_voltage = (
+                "+5V voltage",
+                get_unaligned_be16(msg, self._device_info["plus_5v_voltage"]) * 1e-2,
+                "V",
+            )
+            sensor_readings.append(plus_5v_voltage)
+
+            # Read +12V voltage rail value
+            plus_12v_voltage = (
+                "+12V voltage",
+                get_unaligned_be16(msg, self._device_info["plus_12v_voltage"]) * 1e-2,
+                "V",
+            )
+            sensor_readings.append(plus_12v_voltage)
+
+        return sensor_readings
+
+    def _get_status_from_hwmon(self):
+        sensor_readings = []
+
+        # Read temp sensor values
+        for idx, temp_sensor_offset in enumerate(self._device_info["temp_sensors"]):
+            temp_sensor_reading = (
+                self._device_info["temp_sensors_label"][idx],
+                self._hwmon.get_int(f"temp{idx + 1}_input") * 1e-3,
+                "°C",
+            )
+            sensor_readings.append(temp_sensor_reading)
+
+        # Read fan speed and related values
+        for idx, fan_sensor_offset in enumerate(self._device_info["fan_sensors"]):
+            fan_speed = (
+                self._device_info["fan_speed_label"][idx],
+                self._hwmon.get_int(f"fan{idx + 1}_input"),
+                "rpm",
+            )
+            sensor_readings.append(fan_speed)
+
+            fan_power = (
+                self._device_info["fan_power_label"][idx],
+                self._hwmon.get_int(f"power{idx + 1}_input") * 1e-6,
+                "W",
+            )
+            sensor_readings.append(fan_power)
+
+            fan_voltage = (
+                self._device_info["fan_voltage_label"][idx],
+                self._hwmon.get_int(f"in{idx}_input") * 1e-3,
+                "V",
+            )
+            sensor_readings.append(fan_voltage)
+
+            fan_current = (
+                self._device_info["fan_current_label"][idx],
+                self._hwmon.get_int(f"curr{idx + 1}_input") * 1e-3,
+                "A",
+            )
+            sensor_readings.append(fan_current)
+
+        # Special-case sensor readings
+        if self._device_info["type"] == self.DEVICE_D5NEXT:
+            # Read +5V voltage rail value
+            plus_5v_voltage = ("+5V voltage", self._hwmon.get_int("in2_input") * 1e-3, "V")
+            sensor_readings.append(plus_5v_voltage)
+
+            if self._hwmon.has_attribute("in3_input"):
```

Comment:
Should a warning be produced if this is not available?

Reply:
Yeah, I think so.  That attribute is useful, and our other drivers emit warnings in similar situations:

```py
_LOGGER.warning('some attributes cannot be read from %s kernel driver', self._hwmon.module)
```

I don't think the warning needs to be more specific than that, even though here it's just one attribute.  In the future there might be more (or at least different) attributes missing, and I don't think most users need to care about the specifics (while this is an useful attribute, everything else is still quite functional without it).

That said, one downside of the non-specific warning is that users will need to check the documentation and/or try `--direct-access` just to see what they might be missing.  But that's a general problem with all of our (hwmon-aware) drivers, so this probably isn't the best place to consider it.

Reply:
Added that.